common/awf_vehicle_rviz_plugin/awf_2d_overlay_vehicle/src/signal_display.cpp

```diff
@@ -0,0 +1,493 @@
+    gear_topic_property_->getTopicStd(),
+    rclcpp::QoS(rclcpp::KeepLast(10)).durability_volatile().reliable(),
+    [this](const autoware_auto_vehicle_msgs::msg::GearReport::SharedPtr msg) {
+      updateGearData(msg);
+    });
+
+  steering_sub_ = rviz_node_->create_subscription<autoware_auto_vehicle_msgs::msg::SteeringReport>(
+    steering_topic_property_->getTopicStd(),
+    rclcpp::QoS(rclcpp::KeepLast(10)).durability_volatile().reliable(),
+    [this](const autoware_auto_vehicle_msgs::msg::SteeringReport::SharedPtr msg) {
+      updateSteeringData(msg);
+    });
+
+  speed_sub_ = rviz_node_->create_subscription<autoware_auto_vehicle_msgs::msg::VelocityReport>(
+    speed_topic_property_->getTopicStd(),
+    rclcpp::QoS(rclcpp::KeepLast(10)).durability_volatile().reliable(),
+    [this](const autoware_auto_vehicle_msgs::msg::VelocityReport::SharedPtr msg) {
+      updateSpeedData(msg);
+    });
+
+  turn_signals_sub_ =
+    rviz_node_->create_subscription<autoware_auto_vehicle_msgs::msg::TurnIndicatorsReport>(
+      turn_signals_topic_property_->getTopicStd(),
+      rclcpp::QoS(rclcpp::KeepLast(10)).durability_volatile().reliable(),
+      [this](const autoware_auto_vehicle_msgs::msg::TurnIndicatorsReport::SharedPtr msg) {
+        updateTurnSignalsData(msg);
+      });
+
+  hazard_lights_sub_ =
+    rviz_node_->create_subscription<autoware_auto_vehicle_msgs::msg::HazardLightsReport>(
+      hazard_lights_topic_property_->getTopicStd(),
+      rclcpp::QoS(rclcpp::KeepLast(10)).durability_volatile().reliable(),
+      [this](const autoware_auto_vehicle_msgs::msg::HazardLightsReport::SharedPtr msg) {
+        updateHazardLightsData(msg);
+      });
+
+  traffic_sub_ = rviz_node_->create_subscription<autoware_perception_msgs::msg::TrafficSignalArray>(
+    traffic_topic_property_->getTopicStd(),
+    rclcpp::QoS(rclcpp::KeepLast(10)).durability_volatile().reliable(),
+    [this](const autoware_perception_msgs::msg::TrafficSignalArray::SharedPtr msg) {
+      updateTrafficLightData(msg);
+    });
+
+  speed_limit_sub_ = rviz_node_->create_subscription<tier4_planning_msgs::msg::VelocityLimit>(
+    speed_limit_topic_property_->getTopicStd(),
+    rclcpp::QoS(rclcpp::KeepLast(10)).durability_volatile().reliable(),
+    [this](const tier4_planning_msgs::msg::VelocityLimit::ConstSharedPtr msg) {
+      updateSpeedLimitData(msg);
+    });
+}
+
+SignalDisplay::~SignalDisplay()
+{
+  std::lock_guard<std::mutex> lock(property_mutex_);
+  overlay_.reset();
+
+  gear_sub_.reset();
+  steering_sub_.reset();
+  speed_sub_.reset();
+  turn_signals_sub_.reset();
+  hazard_lights_sub_.reset();
+  traffic_sub_.reset();
+
+  steering_wheel_display_.reset();
+  gear_display_.reset();
+  speed_display_.reset();
+  turn_signals_display_.reset();
+  traffic_display_.reset();
+  speed_limit_display_.reset();
+
+  delete property_width_;
+  delete property_height_;
+  delete property_left_;
+  delete property_top_;
+  delete property_signal_color_;
```

Comment:
[nits] These properties are automatically released by QT's object tree. Since this is the case in overlay_text_display, I think it would be better to delete it so that it becomes the same.

Reply:
Sure thing, will be committing them out of this